### ISSUE TITLE
Undercurl not working - differnt behavior in shell and tmux/neovim

### ISSUE THREAD

**Issue by ziombo** [wez/wezterm#4783]
### What Operating System(s) are you seeing this problem on?

Windows

### WezTerm version

20230817-092749-87a00992

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

There is no constant undercurl behavior between different 'programs'. I followed instructions from https://github.com/wez/wezterm/issues/415 and https://wezfurlong.org/wezterm/faq.html#i-have-set-convert-meta-on-in-my-inputrc-and-latin-characters-are-broken and these don't produce the desired outcome (working undercurl) and also behave differently in different environments. 

I'm running WSL2 inside wezterm.

In ZSH running the line 
`https://wezfurlong.org/wezterm/faq.html#i-have-set-convert-meta-on-in-my-inputrc-and-latin-characters-are-broken`
from the faq produces no underlines/curles at all:
![image](https://github.com/wez/wezterm/assets/15613606/156b24a8-cf60-4e12-9aae-69e47f422202)

When I enter tmux I get the just underline for every part:
![image](https://github.com/wez/wezterm/assets/15613606/8aae9ef7-f0be-4a80-91e0-073726a3ddaf)

Then inside neovim, I'm assuming it's same as for tmux, except due to syntax highlighting each part of underline has same color as the text (I'm guessing the guisp part of highlight group is not applied here):
![image](https://github.com/wez/wezterm/assets/15613606/ae41fda2-a1f5-4f87-abd5-ad5aacb9a1c5)
highlight settings:
![image](https://github.com/wez/wezterm/assets/15613606/518034b9-b935-45d4-8466-524ddb60ec5a)
so if guisp was applied correctly the whole undercurl should have pink-ish color.

I am on windows so possibly this part applies to me

> Note: on Windows, the ConPTY layer strips out the curly underline escape sequences. If you're missing this feature in your WSL instance, you will need to use either wezterm ssh or [multiplexing](https://wezfurlong.org/wezterm/multiplexing.html#connecting-into-windows-subsystem-for-linux) to bypass ConPTY.

But I'm not sure how can I check if it applies to me as the behavior is different depending where I'm trying to show the undercurls.
If that's the case, what does it mean that I have to use wezterm ssh? Do I have to run wsl separately and ssh into it from wezterm?

### Configuration

I tested with and without config and results are same

**Comment by wez** [comment]
> If that's the case, what does it mean that I have to use wezterm ssh? Do I have to run wsl separately and ssh into it from wezterm?

Yeah, if you want to use escape sequences that are not supported by the conpty layer then you'll need to bypass it. The simplest way to do this is to `wezterm ssh` into your WSL or remote linux instance.

You can also use ssh domains together with setting the default domain to make this more automatic and convenient.
Please search the docs and/or issues/discussions for more information on how to do this.
